scripts/test_commit_range.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/sh
+###########################################################################
+#
+#  Copyright 2016-2018, ARM Limited, All Rights Reserved
+#  SPDX-License-Identifier: Apache-2.0
+#
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+###########################################################################
+
+# Check if all commits since a specified commit (not including that commit)
+# build and pass some basic checks.
+
+ORIGIN=origin/development
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
+STATUS=0
+
+usage() {
+    echo "Usage: ./$(basename $0) [<origin-branch>]"
+    echo "    <origin-branch> The branch to test back to (defaults to $ORIGIN)"
+}
+
+if test $# -gt 1; then
+    # There are too many arguments. Show help and exit with an error.
+    usage
+    exit 1
+elif test $# -ge 1; then
+    # The optional <origin-branch> argument is present. Replace the default
+    # value of ORIGIN with the <origin-branch> argument.
+    ORIGIN=$1
+fi
+
+# Check for detached head
+if [ "$BRANCH" = "HEAD" ]; then
+    echo "Please make a branch to run this script from. This script moves"
+    echo "around in history and needs a way back to the starting point."
+    exit 1
+fi
+
+# Check for modified files
+DIFF=$(git diff)
+if [ "$DIFF" != "" ]; then
+    echo "You have modified files. Those changes will be lost when running"
+    echo "this script, so please commit or stash them first."
+    exit 1
+fi
+
+# Check for untracked files
+MODIFIED=$(git status | grep 'Untracked')
+if [ "$MODIFIED" != "" ]; then
+    echo "You currently have untracked files. Untracked files may interfere"
```

Comment:
Seems difficult to add a check for untracked files that have been deleted. We could add that later and remove this check. Without such a check, the script will encounter a failure in git checkout, which is not the greatest user experience, but probably acceptable in a developer script. We should still return to the original commit where the developer started the test from, so the `git checkout` shouldn't be part of `set -e` unless there is some way to set an error handler or something that the script goes back to instead of exiting.